app/Services/AirportService.php

```diff
@@ -46,12 +46,32 @@ public function getMetar($icao)
             return;
         }
 
-        $raw_metar = $this->metarProvider->get_metar($icao);
+        $raw_metar = $this->metarProvider->metar($icao);
         if ($raw_metar && $raw_metar !== '') {
             return new Metar($raw_metar);
         }
     }
 
+    /**
+     * Return the METAR for a given airport
+     *
+     * @param $icao
+     *
+     * @return Metar|null
+     */
+    public function getTaf($icao)
+    {
+        $icao = trim($icao);
+        if ($icao === '') {
+            return;
+        }
+
+        $raw_taf = $this->metarProvider->taf($icao);
+        if ($raw_taf && $raw_taf !== '') {
+            return new Metar($raw_taf, true);
```

Comment:
So I'm a little torn on this. This should be returned but it hasn't been tested very much and will need to be. And tests have to be added for this. Would you be able to put sample TAF strings into this PR? I just need to make sure things don't break before merging this.

Reply:
Here are your sample TAF reports,

LTAI 201040Z 2012/2112 VRB02KT CAVOK BECMG 2012/2015 FEW030 SCT200 =

LTBA 201040Z 2012/2118 04009KT 9999 BKN035 PROB30 TEMPO 2018/2022 -RA FEW015 BKN030 =

LTCC 201040Z 2012/2112 VRB02KT 9999 FEW040 SCT100 BECMG 2017/2019 2500 BR NSC BECMG 2019/2021 1200 BCFG BECMG 2021/2023 0200 FG VV002 BECMG 2105/2106 2000 BCFG TEMPO 2106/2108 0200 FG VV002 BECMG 2108/2111 9999 NSW SCT040 BKN100 =

EDDF 201418Z 2014/2118 06004KT 8000 FEW008 BKN015 TEMPO 2014/2020 OVC008 PROB30 TEMPO 2014/2020 4000 -RADZ BKN004 BECMG 2016/2018 20005KT PROB30 TEMPO 2020/2106 4000 BR BKN004 TEMPO 2111/2118 14008KT TEMPO 2115/2118 RA =

LOWW 201415Z 2015/2121 14010KT 6000 OVC007
    TX04/2113Z TN02/2107Z
    TEMPO 2015/2101 2500 -DZ OVC003
    BECMG 2101/2104 29005KT 2000 OVC003
    PROB40 TEMPO 2104/2109 0500 FG BKN001
    TEMPO 2109/2113 SCT005 SCT040
    BECMG 2113/2115 14007KT 7000 BKN012 =

KBOS 201525Z 2015/2118 19003KT P6SM SCT025 BKN100
     FM201800 18006KT P6SM VCSH BKN025
     FM202000 19007KT P6SM VCSH OVC015
     FM210200 21006KT P6SM OVC015
     FM210500 22004KT P6SM OVC020
     FM210900 36003KT P6SM OVC010 =

CYYZ 201438Z 2015/2118 19008KT 5SM -DZ BR OVC007 TEMPO 2015/2016
     P6SM NSW OVC010
    FM201600 23012KT P6SM OVC015 TEMPO 2016/2018 5SM -DZ BR OVC008
    FM201800 23012G22KT P6SM OVC020
    FM210400 24012KT P6SM BKN020 BKN040
     BECMG 2106/2108 22006KT
    FM211500 16005KT P6SM OVC009
    RMK NXT FCST BY 201800Z = 

Hope these are enough for your tests. (No formatting done by me, all are as they are published by the airport wx offices)